core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1967,6 +2004,23 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
     (scalaTypeVal, SnappySession.getDataType(storeType, storePrecision, storeScale))
   }
+
+
+  SnappyContext.getClusterMode(_sc) match {
+    case ThinClientConnectorMode(_, _) => {
+      _sc.listenerBus.addListener(new AppSparkListener)
+    }
+    case _ => // Nothing to do
+  }
+}
+
+// A listener to let the session know if at least one executor has been
+// added to this application.
+private class AppSparkListener extends SparkListener with Logging {
+  override def onExecutorAdded(execList: SparkListenerExecutorAdded): Unit = {
+    logInfo(s"AppSparkListener: onExecutorAdded: added $execList")
+    SnappySession.executorAssigned = true
+  }
 }
```

Comment:
There is already a SparkContextListener in object SnappyContext. This method can be moved there and there is no need to add another listener in every session.